test/support/generator/create-deal-investor-generator.ts

```diff
@@ -0,0 +1,77 @@
+import { UkefId } from '@ukef/helpers';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { AcbsCreateDealInvestorRequest } from '@ukef/modules/deal-investor/dto/acbs-create-deal-investor-request.dto';
+import { CreateDealInvestorRequest, CreateDealInvestorRequestItem } from '@ukef/modules/deal-investor/dto/create-deal-investor-request.dto';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class CreateDealInvestorGenerator extends AbstractGenerator<CreateDealInvestorRequestItem, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): CreateDealInvestorRequestItem {
+    return {
+      dealIdentifier: this.valueGenerator.ukefId(),
+      portfolioIdentifier: this.valueGenerator.string({ minLength: 0, maxLength: 2 }),
+      lenderType: this.valueGenerator.string({ minLength: 0, maxLength: 3 }),
+      effectiveDate: this.valueGenerator.dateOnlyString(),
+      expiryDate: this.valueGenerator.dateOnlyString(),
+      dealStatus: this.valueGenerator.string({ minLength: 0, maxLength: 1 }),
+      currency: this.valueGenerator.string({ minLength: 0, maxLength: 3 }),
+    };
+  }
+
+  protected transformRawValuesToGeneratedValues(values: CreateDealInvestorRequest, { dealIdentifier, portfolioIdentifier }: GenerateOptions): GenerateResult {
+    const firstDealInvestor = values[0];
+
+    const acbsRequestBodyToCreateDealInvestor: AcbsCreateDealInvestorRequest = {
+      SectionIdentifier: PROPERTIES.DEAL_INVESTOR.DEFAULT.sectionIdentifier ?? '00',
+      EffectiveDate: this.dateStringTransformations.getEffectiveDate(firstDealInvestor.effectiveDate),
+      ExpirationDate: this.dateStringTransformations.getExpirationDate(firstDealInvestor.expiryDate),
+      IsExpirationDateMaximum: firstDealInvestor.expiryDate ? false : true,
+      LenderType: {
+        LenderTypeCode: firstDealInvestor.lenderType ?? PROPERTIES.DEAL_INVESTOR.DEFAULT.lenderType.lenderTypeCode ?? '100',
+      },
+      InvolvedParty: {
+        PartyIdentifier: PROPERTIES.DEAL_INVESTOR.DEFAULT.involvedParty.partyIdentifier ?? '00000000',
+      },
+      Currency: {
+        CurrencyCode: firstDealInvestor.currency,
+      },
+      CustomerAdvisedIndicator: PROPERTIES.DEAL_INVESTOR.DEFAULT.customerAdvisedIndicator ?? false,
+      DealStatus: {
+        DealStatusCode: firstDealInvestor.dealStatus ?? PROPERTIES.DEAL_INVESTOR.DEFAULT.dealStatus.dealStatusCode ?? 'A',
+      },
+      UserDefinedCode1: PROPERTIES.DEAL_INVESTOR.DEFAULT.userDefinedCode1 ?? 'N',
+      ContractPercentage: PROPERTIES.DEAL_INVESTOR.DEFAULT.contractPercentage,
+      LimitRevolvingIndicator: PROPERTIES.DEAL_INVESTOR.DEFAULT.limitRevolvingIndicator ?? false,
```

Comment:
We don't need the `?? <hard-coded value>` bits here

Reply:
Done